linked_list/linked_list.py

```diff
@@ -23,13 +23,70 @@ def __init__(self):
     self.tail = None
 
   def add_to_tail(self, value):
-    pass
+    new = Node(value)
+    if self.tail is None:
+      self.tail  = new
+      self.head = new
+    elif self.head == self.tail:
+      self.head.set_next(new)
+      self.tail = new
+    else:
+      self.tail.set_next(new)
+      self.tail = new
 
   def remove_head(self):
-    pass
+    if self.head is None:
+      print("no head")
+    else:
+      headprev = self.head.get_value()
+      if self.head.get_next() is None:
+        self.head = None
+        self.tail = None
+        return headprev
+      else:
+        self.head = self.head.get_next()
+        print(headprev)
+        return headprev
 
-  def contains(self, value):
-    pass
+  # def contains(self, value):
+  #   current = self.head
+  #   match = False
+  #   if current is None:
+  #     return False
+  #   elif current.get_value() == value:
+  #     return True
+  #   else:
+  #     while match == False:
+  #       if current.get_next() == None:
+  #         return False
+  #       else:
+  #         current = current.get_next()
+  #         if current.get_value() == value:
+  #           match = True
+  #           return True
+
+  def contains(self, value, current="koolkats"):
+    if current == "koolkats":
+      current = self.head
+    if current == None:
+      return False
+    elif current.get_next() == None and current.get_value() != value:
+      return False
+    elif current.get_value() != value:
+      return self.contains(value, current.get_next())
+    else:
+      return True
 
   def get_max(self):
-    pass
+    if self.head is None:
+      return None
+    else:
+      current = self.head
+      max = self.head.get_value()
```

Comment:
Just a tiny thing, `max` is the name of the built in function that finds the largest element in a list or another iterable. By assigning `self.head.get_value()`, you are overwriting the base functionality within this scope. 

Since you're not using `max` in this block, no problem. But you wanna be aware of it; It may cause problems in the future.